pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -382,3 +399,48 @@ func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
 		c.queue.Forget(key)
 	}
 }
+
+// cleanupOldDeployments triggers on deployment configuration update and add events in order
+// to delete old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeployments []kapi.ReplicationController, deploymentConfig *deployapi.DeploymentConfig) error {
+	if deploymentConfig.Spec.RevisionHistoryLimit == nil {
+		// there is no past deplyoment quota set
+		return nil
+	}
+
+	activeDeployment := deployutil.ActiveDeployment(deploymentConfig, existingDeployments)
```

Comment:
What happens if I have cleanup policy set but still haven't deployed successfully ie. all of my deployments have failed so far? There is no active deployment in that case.